### PR TITLE
fix: No chapters flag not respected

### LINKED ISSUE #2500: No chapters flag not respected
**What version of EMX2 are you using (see footer)**
v8.200.2

**Describe the bug**
isChaptersEnabled is set to false, but row editor still shows chapters

### LINKED ISSUE #2501: Row edit modal is broken ( cannot select next chapter )
**What version of EMX2 are you using (see footer)**
8.200.2

**Describe the bug**
When selecting a chapter from the chapter side menu nothing happens

**To Reproduce**
1. go to catalogue schema , sign in with edit permission 
2. edit a cohort row 
3. try next chapter  -> nothing happens
4. try using side panel -> nothing happens

**Expected behavior**
next chapter is shown

### PR DESCRIPTION
**Pull request by connoratrug**
- Respect feature flag
- Allow passing flag instead of fetching
- Extend demo component to toggle feature flag on and off

Closes #2500
Closes #2501

### CODE REVIEW

File: apps/molgenis-components/src/components/forms/EditModal.vue [hunk 1]
```diff
@@ -134,9 +134,9 @@ export default {
       schemaMetaData: null as unknown as ISchemaMetaData,
       client: null as unknown as INewClient,
       errorMessage: "",
-      loaded: true,
+      loaded: false,
       currentPage: 1,
-      useChapters: true,
+      myUseChapters: this.useChapters,
```
Comment (mswertz) [inline apps/molgenis-components/src/components/forms/EditModal.vue:139]: we might need a better name. Something like 'hasUseChapters' or something?

File: apps/molgenis-components/src/components/forms/EditModal.vue [hunk 2]
```diff
@@ -285,10 +292,12 @@ export default {
     this.schemaMetaData = await this.client.fetchSchemaMetaData();
     const settings: ISetting[] = await this.client.fetchSettings();
 
-    this.useChapters =
-      settings.find(
-        (item: ISetting) => item.key === IS_CHAPTERS_ENABLED_FIELD_NAME
-      )?.value !== "false";
+    if (this.useChapters === null) {
+      this.myUseChapters =
```
Comment (mswertz) [inline apps/molgenis-components/src/components/forms/EditModal.vue:296]: Ah, I  now understand. It is the mix between the prop and the settings

File: apps/molgenis-components/src/components/forms/EditModal.vue (hunk 1 above)
Comment (mswertz) [reply apps/molgenis-components/src/components/forms/EditModal.vue:139]: probably good to look for a convention of this, the combo of prop vs server backed state.

**Review by mswertz — APPROVED** [review]

File: apps/molgenis-components/src/components/forms/EditModal.vue (hunk 2 above)
Comment (connoratrug) [reply apps/molgenis-components/src/components/forms/EditModal.vue:296]: yes , the my.. is old school java naming convention ( js used to use _foo ) to my knowledge js/vue does not have convention  for this as injection the data via prop in way data is passed in most examples i see

File: apps/molgenis-components/src/components/forms/EditModal.vue (hunk 1 above)
Comment (connoratrug) [reply apps/molgenis-components/src/components/forms/EditModal.vue:139]: there is no real convention that i know, vue 2 used to use thisValue to break the reactive chain between prop and reactive mutable value

**Pull request merged**